### PR TITLE
[bitnami/postgresql] - Allow configure replication for postgresql chart 

### PR DESCRIPTION
**Pull request by tompizmor**
This PRs add support to the postgresql chart to configures a replication master-slave. 
It also add other improvements like:
- configurable readiness and liveness probes
- ability to enable/disable a metrics exporter
- ability to mount custom config file via config map

### CODE REVIEW

File: bitnami/postgresql/templates/NOTES.txt [hunk 1]
```diff
@@ -1,8 +1,54 @@
+{{- if contains .Values.service.type "LoadBalancer" }}
+{{- if not .Values.postgresPassword }}
```
Comment (juan131) [inline bitnami/postgresql/templates/NOTES.txt]: Typo. Use `.Values.postgresqlPassword`

File: bitnami/postgresql/templates/secrets.yaml [hunk 2]
```diff
@@ -1,16 +1,23 @@
...
-    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    app: {{ template "postgresql.name" . }}
+    chart: {{ template "postgresql.chart" . }}
     release: "{{ .Release.Name }}"
     heritage: "{{ .Release.Service }}"
 type: Opaque
 data:
   {{ if .Values.postgresqlPassword }}
-  postgresql-password: {{ default "" .Values.postgresqlPassword | b64enc | quote }}
+  postgresql-password: {{ .Values.postgresqlPassword | b64enc | quote }}
```
Comment (juan131) [inline bitnami/postgresql/templates/secrets.yaml]: What happens if `.Values.postgresqlPassword` is not defined?

File: bitnami/postgresql/templates/statefulset.yaml [hunk 3]
```diff
@@ -0,0 +1,148 @@
+apiVersion: apps/v1beta1
```
Comment (juan131) [inline bitnami/postgresql/templates/statefulset.yaml]: I see different solutions for the StatefulSet API version.. Some charts use `apps/v1beta1`, others `apps/v1beta2` and others `apps/v1`. We should standardise it.

File: bitnami/postgresql/templates/svc.yaml [hunk 4]
```diff
@@ -1,17 +1,19 @@
 apiVersion: v1
```
Comment (juan131) [inline bitnami/postgresql/templates/svc.yaml]: Should we call it `svc-master.yaml`?

File: bitnami/postgresql/files/README.md [hunk 5]
```diff
@@ -0,0 +1 @@
+Copy here your postgresql.conf file to use it as a config map.
```
[the remark this answers is not in the document]
Comment (tompizmor) [reply bitnami/postgresql/files/README.md]: ok

File: bitnami/postgresql/templates/NOTES.txt (hunk 1 above)
Comment (tompizmor) [reply bitnami/postgresql/templates/NOTES.txt]: thanks

File: bitnami/postgresql/templates/secrets.yaml (hunk 2 above)
Comment (tompizmor) [reply bitnami/postgresql/templates/secrets.yaml]: Then a 10 character alphanumeric password is generated:

```
  {{ if .Values.postgresqlPassword }}
  postgresql-password: {{ .Values.postgresqlPassword | b64enc | quote }}
  {{ else }}
  postgresql-password: {{ randAlphaNum 10 | b64enc | quote }}
  {{ end }}
```

File: bitnami/postgresql/templates/statefulset.yaml (hunk 3 above)
Comment (tompizmor) [reply bitnami/postgresql/templates/statefulset.yaml]: You are right! I think we should use `apps/v1beta2`. 
The `apps/v1` only works in very new kubernetes clusters and we have already had issues in the past with this API version. https://github.com/kubernetes/charts/issues/4908

File: bitnami/postgresql/templates/svc.yaml (hunk 4 above)
Comment (tompizmor) [reply bitnami/postgresql/templates/svc.yaml]: I prefer to keep it just `svc.yaml` because it also applies when you deploy a single postgresql node.

**Pull request merged**